src/DIRAC/WorkloadManagementSystem/Agent/PilotLoggingAgent.py

```diff
@@ -0,0 +1,148 @@
+
+
+class PilotLoggingAgent(AgentModule):
+    """
+    .. class:: PilotLoggingAgent
+
+    The agent sends completed pilot log files to permanent storage for analysis.
+    """
+
+    def initialize(self):
+        """
+        agent's initalisation. Use this agent's CS information to:
+        Determine what Defaults/Shifter shifter proxy to use.,
+        get the target SE name from the CS.
+        Obtain log file location from Tornado.
+
+        :param self: self reference
+        """
+
+        # configured VOs and setup
+        res = getVOs()
+        if res["OK"]:
+            self.voList = res.get("Value", [])
+        else:
+            return S_ERROR(res["Message"])
+
+        if isinstance(self.voList, str):
+            self.voList = [self.voList]
+
+        self.setup = gConfig.getValue("/DIRAC/Setup", None)
+        if self.setup is None:
+            self.log.error("Setup is not defined in the configuration")
+            return S_ERROR("Setup is not defined in the configuration")
+
+        return S_OK()
+
+    def execute(self):
+        """
+        Execute one agent cycle. Upload log files to the SE and register them in the DFC.
+        Use a shifter proxy dynamically loaded for every VO
+
+        :param self: self reference
+        """
+        voRes = {}
+        for vo in self.voList:
+            self.opsHelper = Operations(vo=vo, setup=self.setup)
+            # is remote pilot logging enabled for the VO ?
+            pilotLogging = self.opsHelper.getValue("/Pilot/RemoteLogging", True)
+            if pilotLogging:
+                res = self.opsHelper.getOptionsDict("/Shifter/DataManager")
+                if not res["OK"]:
+                    voRes[vo] = "No shifter defined - skipped"
+                    self.log.error(f"No shifter defined for VO: {vo} - skipping ...")
+                    continue
+
+                proxyUser = res["Value"].get("User", None)
+                proxyGroup = res["Value"].get("Group", None)
+                if proxyGroup is None or proxyUser is None:
+                    self.log.error(
+                        f"No proxy user or group defined for pilot: VO: {vo}, User: {proxyUser}, Group: {proxyGroup}"
+                    )
+                    voRes[vo] = "No proxy user or group defined - skipped"
+                    continue
+
+                self.log.info(f"Proxy used for pilot logging: VO: {vo}, User: {proxyUser}, Group: {proxyGroup}")
+                res = self.executeForVO(vo, proxyUserName=proxyUser, proxyUserGroup=proxyGroup)
+                if not res["OK"]:
+                    voRes[vo] = res["Message"]
+        if voRes:
+            voRes.update(S_ERROR("Agent cycle for some VO finished with errors"))
+            return voRes
+        return S_OK()
+
+    @executeWithUserProxy
+    def executeForVO(self, vo):
+        """
+        Execute one agent cycle for VO
+
+        :param vo: vo enabled for remote pilot logging
+        :type vo: str
```

Comment:
```suggestion
        :param str vo: vo enabled for remote pilot logging
```
But using type annotations would be best